scalafix-rules/src/main/scala-3/scalafix/internal/rule/Embedded.scala

```diff
@@ -0,0 +1,94 @@
+package scalafix.internal.rule
+
+import java.net.URLClassLoader
+import java.nio.file.Path
+import java.util.ServiceLoader
+
+import scala.jdk.CollectionConverters.*
+
+import scala.meta.pc.PresentationCompiler
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import coursierapi.MavenRepository
+
+object Embedded {
+
+  def presentationCompiler(
+      scalaVersion: String
+  ): PresentationCompiler = {
+    val deps =
+      scala3PresentationCompilerDependencies(scalaVersion)
+    val jars = Fetch
+      .create()
+      .addDependencies(deps*)
+      .addRepositories(
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/snapshots"
+        )
+      )
+      .fetch()
+      .asScala
+      .map(_.toPath())
+      .toSeq
+    val classloader = newPresentationCompilerClassLoader(jars)
+
+    val presentationCompilerClassname =
+      if (supportPresentationCompilerInDotty(scalaVersion)) {
+        "dotty.tools.pc.ScalaPresentationCompiler"
+      } else {
+        "scala.meta.pc.ScalaPresentationCompiler"
+      }
+
+    serviceLoader(
+      classOf[PresentationCompiler],
+      presentationCompilerClassname,
+      classloader
+    )
+  }
+
+  private def supportPresentationCompilerInDotty(scalaVersion: String) = {
+    scalaVersion.split("\\.").take(3).map(_.toInt) match {
+      case Array(3, minor, patch) => minor > 3 || minor == 3 && patch >= 4
+      case _ => false
+    }
+  }
```

Comment:
Since we control the runtime of scalafix, and we'll first [publish](https://github.com/scalacenter/scalafix/issues/1680) scala3 artifacts with [3.3.4 or later](https://github.com/scalacenter/scalafix/pull/1926), this should always be true. If that's correct, I assume we don't need coursier and we can declare the dependency statically in scalafix-rules_3?